src/ngAnimate/animateCss.js

```diff
@@ -0,0 +1,670 @@
+// therefore there is no reason to test anymore for other vendor prefixes:
+// http://caniuse.com/#search=transition
+if (window.ontransitionend === undefined && window.onwebkittransitionend !== undefined) {
+  CSS_PREFIX = '-webkit-';
+  TRANSITION_PROP = 'WebkitTransition';
+  TRANSITIONEND_EVENT = 'webkitTransitionEnd transitionend';
+} else {
+  TRANSITION_PROP = 'transition';
+  TRANSITIONEND_EVENT = 'transitionend';
+}
+
+if (window.onanimationend === undefined && window.onwebkitanimationend !== undefined) {
+  CSS_PREFIX = '-webkit-';
+  ANIMATION_PROP = 'WebkitAnimation';
+  ANIMATIONEND_EVENT = 'webkitAnimationEnd animationend';
+} else {
+  ANIMATION_PROP = 'animation';
+  ANIMATIONEND_EVENT = 'animationend';
+}
+
+var DURATION_KEY = 'Duration';
+var PROPERTY_KEY = 'Property';
+var DELAY_KEY = 'Delay';
+var TIMING_KEY = 'TimingFunction';
+var ANIMATION_ITERATION_COUNT_KEY = 'IterationCount';
+var ANIMATION_PLAYSTATE_KEY = 'PlayState';
+var ELAPSED_TIME_MAX_DECIMAL_PLACES = 3;
+var CLOSING_TIME_BUFFER = 1.5;
+var ONE_SECOND = 1000;
+var BASE_TEN = 10;
+
+function computeCSSStyles($window, element, properties) {
+  var styles = {};
+
+  var detectedStyles = $window.getComputedStyle(element) || {};
+  forEach(properties, function(style, prop) {
+    var val = detectedStyles[style];
+    // only numerical-based values have a digit as the first value
+    if (val) {
+      if (val.charAt(0) >= 0) {
+        val = parseMaxTime(val);
+      }
+      styles[prop] = val;
+    }
+  });
+
+  return styles;
+}
+
+function parseMaxTime(str) {
+  var maxValue = 0;
+  var values = isString(str) ?
+    str.split(/\s*,\s*/) :
+    [];
+  forEach(values, function(value) {
+    maxValue = Math.max(parseFloat(value) || 0, maxValue);
+  });
+  return maxValue;
+}
+
+function getCssTransitionDurationStyle(duration, applyOnlyDuration) {
+  var style = TRANSITION_PROP;
+  var value = duration + 's';
+  if (applyOnlyDuration) {
+    style += DURATION_KEY;
+  } else {
+    value += ' linear all';
+  }
+  return [style, value];
+}
+
+function getCssKeyframeDurationStyle(duration) {
+  return [ANIMATION_PROP + DURATION_KEY, duration + 's'];
+}
+
+function getCssDelayStyle(delay, isKeyframeAnimation) {
+  var prop = (isKeyframeAnimation ? ANIMATION_PROP : TRANSITION_PROP) + DELAY_KEY;
+  return [prop, delay + 's'];
+}
+
+function blockTransitions(node, bool) {
+  var value = bool ? 'none' : '';
+  var key = TRANSITION_PROP + PROPERTY_KEY;
+  applyInlineStyle(node, key, value);
+  return [key, value];
+}
+
+function blockAnimations(node, bool) {
+  var value = bool ? 'paused' : '';
+  var key = ANIMATION_PROP + ANIMATION_PLAYSTATE_KEY;
+  applyInlineStyle(node, key, value);
+  return [key, value];
+}
+
+function applyInlineStyle(node, key, value) {
+  if (arguments.length == 2) {
+    value = key[1];
+    key = key[0];
+  }
+  node.style[key] = value;
+}
+
+function LocalCacheLookup() {
```

Comment:
Since this is really a factory function, I would rename it to `createLocalCacheLookup()` and then don't instantiate it with `new` below.